spec/models/ability_spec.rb

```diff
@@ -1,437 +1,240 @@
-          end
+      describe "Student" do
+        let(:student_team) { build(:team, :in_current_season) }
+        let(:second_team) { build(:team, :in_current_season) }
+        before do
+          allow(user).to receive(:current_student?).and_return(true)
         end
+        # TODO Question: is this ^ stub correct to test the ability only?
 
-      end
-
-      describe 'who is disallowed to see email address in user profile' do
+        it_behaves_like "same public features as confirmed user"
+        it { expect(subject).to be_able_to(:create, Conference) }
 
-        # email address is hidden: not admin, not user's supervisor in current season
-        # email address is not hidden: not admin, not confirmed user, not user him/herself
-
-        let(:other_user) { create(:user) }
-
-        context 'when email address is hidden' do
-          context "when not an admin or user's supervisor in current season" do
-            before do
-              allow(user).to receive(:admin?).and_return(false)
-              allow(ability).to receive(:supervises?).with(other_user, user).and_return(false)
-            end
-            it 'disallows to see hidden email address' do
-              other_user.hide_email = true
-              expect(ability).not_to be_able_to(:read_email, other_user)
-            end
-          end
+        it "can create their first team" do
+          expect(subject).to be_able_to(:create, student_team)
         end
-
-        context 'when email address is not hidden' do
-          context 'when not an admin or confirmed user or user him/herself' do
-            before do
-              allow(user).to receive(:admin?).and_return(false)
-              allow(user).to receive(:confirmed?).and_return(false)
-            end
-            it 'disallows to see not hidden email address' do
-              other_user.hide_email = false
-              expect(ability).not_to be_able_to(:read_email, other_user)
-            end
-          end
+        it 'cannot create a second team in current_season' do
+          pending 'fails; it is tested in team_spec.rb:33, and that passes'
+          expect(subject).not_to be_able_to(:create, second_team)
         end
 
+        let(:future_team) { build(:team, season: Season.succ ) }
+        it { expect(subject).to be_able_to :create, future_team }
       end
 
-
-      context 'resend_confirmation_instruction' do
+      describe "Supervisor" do
         let(:other_user) { create(:user) }
+        let(:other_team_user) { build_stubbed(:user) }
 
-        describe 'a user can only resend her/his own confirmation' do
-          it { expect(ability).to be_able_to(:resend_confirmation_instruction, user) }
-          it { expect(ability).not_to be_able_to(:resend_confirmation_instruction, other_user) }
+        before do
+          allow(user).to receive(:supervisor?).and_return(true)
+          allow(ability).to receive(:supervises?) { false } # Rspec complained: "stub default value first"
```

Comment:
I sorta understand why RSpec does this 😉 

I'd personally never use the `receive(:thing).with` when defining a mock (with `allow`). The matcher / syntax makes way more sense for **expecting message calls** in my opinion.
So within the test you'd have something like
```ruby
expect(model).to receive(:thing?).with(other_thing).and_return(false)
```
☝️ testing the "contract" so to say.

Reply:
I had to look up the difference, but I still don't understand what it is happening. 
If I change the line to 
  ```expect(ability).to receive(:supervises?).with(other_user, user) { true }```
it fails with `received 0 times`
Do you understand what I am missing here?

Reply:
Oh sorry, I did not mean you should change it!

Expecting a method call only makes sense (and will only pass) within an example. So if your test want to check whether or not the specific message is passed to this object but you don't care about the inner logic and want to stub it, it may look like this:
```ruby
it 'is able to do something' do
  expect(ability).to receive(:supervise?).with(other_user, user).and_return(true)
  expect(ability).to be_able_to(:read, :users_info, other_user)
end
```

Reply:
OOOOH... and you actually said that: "within an example". Sorry.
NOW I understand the fail message "provide a stub first" 💃